src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -2023,11 +2049,36 @@ public override BoundStatement BindStatement(StatementSyntax node, DiagnosticBag
 
                     return statement;
                 }
-                else
+
+                return (BoundStatement)boundNode;
+            }
+
+            private BoundNode TryGetBoundNodeFromMap(CSharpSyntaxNode node)
+            {
+                ImmutableArray<BoundNode> boundNodes = _semanticModel.GuardedGetBoundNodesFromMap(node);
+
+                if (!boundNodes.IsDefaultOrEmpty)
                 {
                     // Already bound. Return the top-most bound node associated with the statement. 
-                    return (BoundStatement)boundNodes[0];
+                    return boundNodes[0];
                 }
+
+                return null;
+            }
+
+            public override BoundNode BindMethodBody(CSharpSyntaxNode node, DiagnosticBag diagnostics)
+            {
+                return TryGetBoundNodeFromMap(node) ?? base.BindMethodBody(node, diagnostics);
```

Comment:
When the bound node was not already in the map, and we decide to bind, are the resulting bound nodes getting saved into the map?

Reply:
> When the bound node was not already in the map, and we decide to bind, are the resulting bound nodes getting saved into the map?

Yes.

---
In reply to: [170084603](https://github.com/dotnet/roslyn/pull/24918#discussion_r170084603) [](ancestors = 170084603)